app/code/Magento/InventoryInStorePickupAdminUi/Controller/Adminhtml/Order/NotifyPickup.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupAdminUi\Controller\Adminhtml\Order;
+
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryInStorePickupApi\Api\NotifyOrderIsReadyAndShipInterface;
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Class NotifyPickup
+ *
+ * @package Magento\InventoryInStorePickupAdminUi\Controller\Adminhtml\Order
+ */
+class NotifyPickup extends \Magento\Backend\App\Action
+{
+    /**
+     * Authorization level of a basic admin session
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Magento_Sales::emails';
+
+    /**
+     * @var \Magento\InventoryInStorePickupApi\Api\NotifyOrderIsReadyAndShipInterface
+     */
+    private $notifyOrderIsReadyAndShip;
+
+    /**
+     * @var \Magento\Sales\Api\OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * NotifyPickup constructor.
+     *
+     * @param \Magento\Backend\App\Action\Context                                       $context
+     * @param \Magento\InventoryInStorePickupApi\Api\NotifyOrderIsReadyAndShipInterface $notifyOrderIsReadyAndShip
+     * @param \Magento\Sales\Api\OrderRepositoryInterface                               $orderRepository
+     * @param \Psr\Log\LoggerInterface                                                  $logger
+     */
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
+        NotifyOrderIsReadyAndShipInterface $notifyOrderIsReadyAndShip,
+        OrderRepositoryInterface $orderRepository,
+        LoggerInterface $logger
+    ) {
+        $this->notifyOrderIsReadyAndShip = $notifyOrderIsReadyAndShip;
+        $this->orderRepository = $orderRepository;
+        $this->logger = $logger;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * Notify user
+     *
+     * @return \Magento\Framework\Controller\ResultInterface
+     */
+    public function execute():\Magento\Framework\Controller\ResultInterface
+    {
+        $order = $this->initOrder();
+
+        if ($order) {
+            try {
+                $this->notifyOrderIsReadyAndShip->execute((int)$order->getEntityId());
+                $this->messageManager->addSuccessMessage(__('The customer have been notified and shipment created.'));
```

Comment:
all new phrases should be added to translation file

Reply:
Done